sheepdog/transactions/upload/sub_entities.py

```diff
@@ -318,14 +278,16 @@ def _set_node_and_file_ids(self):
         index service.
         """
         if not self.file_exists:
-            if self.entity_id:
-                # use entity_id for file creation
-                self.file_index = self.entity_id
-            else:
-                # use same id for both node entity id and file index
+            # generate entity_id if not provided
+            if not self.entity_id:
                 self.entity_id = str(uuid.uuid4())
-                self.file_index = self.entity_id
+            # use same id for both node entity id and file index
+            self.file_index = self.entity_id
         else:
+            # If hash uniqueness not enforced, do nothing
+            if not flask.current_app.config.get('ENFORCE_FILE_HASH_UNIQUENESS', True):
```

Comment:
this config should be documented in dev_settings and describe the behavior is it's set to False